cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -381,11 +381,21 @@ class HydratorPlusPlusTopPanelCtrl {
             - Multiple macros (e.g. '${macro1}${macro2}')
             - And combined (e,g, '${function(${macro1})}${macro2}')
             More complicated cases will be handled by the backend
+
+            TODO: CDAP-17726 - We need API from backend to get macros, given a pipeline config
+            The logic UI uses to parse and understand a macro is faulty and does not cover
+            complex cases while running preview.
+
+            This is a temporary fix to not surface simple cases of macro functions.
+            Hence the specific check for known macro functions that doesn't need user input.
           */
           if (macroString &&
             typeof macroString === 'string' &&
             macroString.indexOf('${') !== -1 &&
-            macroString.indexOf('}') !== -1
+            macroString.indexOf('}') !== -1 &&
+            macroString !== '${logicalStartTime(}' &&
```

Comment:
Is this correct?

Reply:
As in, should this be `!==` or `!macroString.startsWith`?

Reply:
Have fixed it..